### PR TITLE
raft_client: connect to other stores using peer_address (#5569)

### PR DESCRIPTION
**Pull request by disksing**
Signed-off-by: disksing <i@disksing.com>

###  What have you changed?
We plan to provide a heterogeneous engine with tikv, which may need to use different addresses to provide kv and raft services. Now kvproto has an optional `peer_address` to support this. 

The corresponding tikv modification is: use `peer_address` when connecting other nodes to send raft messages. (for backward compatibility: if `peer_address` is not set, `address` is still used)

###  What is the type of the changes?
- New feature (a change which adds functionality)

###  How is the PR tested?
- Unit test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?
No

###  Does this PR affect `tidb-ansible`?
No

###  Refer to a related PR or issue link (optional)
cherry-pick: #5569

### CODE REVIEW

File: src/server/resolve.rs [hunk 1]
```diff
@@ -69,14 +69,14 @@ impl<T: PdClient> Runner<T> {
...
-        let s = box_try!(pd_client.get_store(store_id));
+        let mut s = box_try!(pd_client.get_store(store_id));
         if s.get_state() == metapb::StoreState::Tombstone {
             RESOLVE_STORE_COUNTER
                 .with_label_values(&["tombstone"])
                 .inc();
             return Err(box_err!("store {} has been removed", store_id));
         }
-        let addr = s.get_address().to_owned();
+        let addr = take_peer_address(&mut s);
```
Comment (siddontang) [inline src/server/resolve.rs]: is it ok to use take here?

Comment (disksing) [reply src/server/resolve.rs]: yes, `s` is never used after this line.

File: Cargo.lock [hunk 2]
```diff
@@ -2228,7 +2228,7 @@ version = "0.4.3"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 dependencies = [
  "fxhash 0.2.1 (registry+https://github.com/rust-lang/crates.io-index)",
- "log 0.4.6 (registry+https://github.com/rust-lang/crates.io-index)",
+ "log 0.3.9 (registry+https://github.com/rust-lang/crates.io-index)",
```
Comment (BusyJay) [inline Cargo.lock]: log is downgraded. I'm afraid it may lose filenames.

**Pull request merged**